ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -64,18 +67,28 @@ function pre-ocp-66839(){
         echo "Failed to extract CRs from live cluster!"
         return 1
     fi
-    if grep -r "capability.openshift.io/name:" "${preCredsDir}"; then
-        echo "Extracted CRs has cap annotation, but expected nothing"
-        return 1
+
+    if [[ "${ADDITIONAL_ENABLED_CAPABILITIES}" != "" ]]; then
+        curCapInCR=$(grep -rh "capability.openshift.io/name:" "${preCredsDir}"|awk -F": " '{print $NF}'|sort -u|xargs)
+        if [[ "${curCapInCR}" != "${expectedCap}" ]]; then
+            echo "Extracted CRs has cap annotation: ${curCapInCR}, but expected ${expectedCap}"
+            return 1
+        fi
+    else
+        if grep -r "capability.openshift.io/name:" "${preCredsDir}"; then
+            echo "Extracted CRs has cap annotation, but expected nothing"
+            return 1
+        fi
     fi
+
     # Extract all CRs from tobe upgrade release payload with --included
     if ! oc adm release extract --to "${tobeCredsDir}" --included --credentials-requests "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"; then
         echo "Failed to extract CRs from tobe upgrade release payload!"
         return 1
     fi
     tobecap=$(grep -rh "capability.openshift.io/name:" "${tobeCredsDir}"|awk -F": " '{print $NF}'|sort -u|xargs)
-    if [[ "${tobecap}" != "MachineAPI ImageRegistry" ]] && [[ "${tobecap}" != "ImageRegistry MachineAPI" ]]; then
-        echo "Tobe gained CRs with cap annotation: ${tobecap}, but expected: MachineAPI and ImageRegistry"
+    if [[ "${tobecap}" != "${EXPECTED_CAPABILITIES_IN_CREDENTIALREQUEST}" ]]; then
```

Comment:
the same comments as https://github.com/openshift/release/pull/44820/files#r1377095365